src/lib.rs

```diff
@@ -260,6 +260,114 @@ where
             base: self.value_map.iter(),
         }
     }
+
+    /// Returns the number of elements the map can hold without reallocating.
+    ///
+    /// This number is a lower bound; the collection might be able to hold
+    /// more, but is guaranteed to be able to hold at least this many.
+    #[inline]
+    pub fn capacity(&self) -> usize {
+        std::cmp::min(self.value_map.capacity(), self.key_map.capacity())
+    }
+
+    /// Clears the map, removing all key-value pairs, just like `HashMap::clear`.
+    /// Keeps the allocated memory for reuse.
+    pub fn clear(&mut self) {
+        self.value_map.clear();
+        self.key_map.clear();
+    }
+
+    /// Clears the map, returning an iterator over the primary key and the
+    /// secondary key + value tuple. Keeps the allocated memory for reuse.
+    ///
+    /// There's no `drain_alt` providing an iterator over secondary key and value,
+    /// but you can map the iterator as follows:
+    /// ```
+    /// use multi_map::MultiMap;
+    ///
+    /// let mut a = MultiMap::new();
+    /// a.insert("apple", 1, "a");
+    /// a.insert("banana", 2, "b");
+    ///
+    /// for (k, v) in a.drain().map(|(k1, (k2, v))| (k2, v)).take(1) {
+    ///     assert!(k == 1 || k == 2);
+    ///     assert!(v == "a" || v == "b");
+    /// }
+    ///
+    /// assert!(a.is_empty());
+    /// ```
+    pub fn drain(&mut self) -> hash_map::Drain<'_, K1, (K2, V)> {
+        self.key_map.clear();
```

Comment:
Shouldn't this be `drain`?
```suggestion
        self.key_map.drain();
```

Reply:
Since one of the maps contains the set of the three values K1, K2, V we return the drain of that one. Draining has the side effect of clearing the map, so for consistency we also must clear our adjacent `key_map`.

As an aside, this is another case where for ergonomics we should have a Drain struct that produces output to match `insert` (a remapped tuple from `(K1, (K2, V))` into  `(K1, K2, V)`), but we'd have to create it from scratch because `hash_map::Drain` isn't directly suitable for this.

Reply:
Ah, I see. Thanks.